File: app/src/main/java/com/example/itcontroller/movies/MainActivity.java
package com.example.itcontroller.movies;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;

import com.example.itcontroller.movies.APIs.API;
import com.example.itcontroller.movies.models.Movie;
import com.example.itcontroller.movies.models.MovieResult;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Log.d("aaa", "Main Activity");
        API.getAPIinstance().getNowPlayingAPI().getNowPlaying().enqueue(new Callback<MovieResult>() {
            @Override
            public void onResponse(Call<MovieResult> call, Response<MovieResult> response) {


                /////////////////////////////////////////////////////////////////////////////////////////////////
                Log.d("aaa", response.body().getResults().length + "");
                for (int i = 0; i < response.body().getResults().length; i++) {
                    Log.d("aaa", "" + response.body().getResults()[i].getAdult());
                    Log.d("aaa", "" + response.body().getResults()[i].getOriginal_language());
                    Log.d("aaa", "" + response.body().getResults()[i].getOriginal_title());
                    Log.d("aaa", "" + response.body().getResults()[i].getPopularity());
                    Log.d("aaa", "=========================================================");
                }
                ///////////////////////////////////////////////////////////////////////////////////////////////////
            }

            @Override
            public void onFailure(Call<MovieResult> call, Throwable t) {
                Log.d("aaa", "failed");
            }
        });
    }
}

File: app/src/main/java/com/example/itcontroller/movies/MainActivity.kt
package com.example.itcontroller.movies

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import com.example.itcontroller.movies.Utils.StringUtils
import com.example.itcontroller.movies.activities.DiscoverMoviesActivity
import com.example.itcontroller.movies.activities.MoviesViewActivity
import kotlinx.android.synthetic.main.activity_main.*
class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        Log.d("aaa", "Main Activity")
        val i :Intent = Intent(this@MainActivity,MoviesViewActivity::class.java)
        var di  = Intent(this@MainActivity,DiscoverMoviesActivity::class.java)
        btnNowPlaying.setOnClickListener {
            i.putExtra(StringUtils.MOVIE_VIEW_TYPE, StringUtils.NOW_PLAYING)
            startActivity(i)
        }
        btnPopular.setOnClickListener {
            i.putExtra(StringUtils.MOVIE_VIEW_TYPE, StringUtils.POPULAR)
            startActivity(i)
        }
        btnTopRated.setOnClickListener {
            i.putExtra(StringUtils.MOVIE_VIEW_TYPE, StringUtils.TOP_RATED)
            startActivity(i)
        }
        btnUpcoming.setOnClickListener {
            i.putExtra(StringUtils.MOVIE_VIEW_TYPE, StringUtils.UP_COMING)
            startActivity(i)
        }
        btnDiscover.setOnClickListener {
            startActivity(di)
        }
    }
}
